### PR TITLE
[DEV] Add the `clean` subcommand.

### PR DESCRIPTION
**Pull request by cpb-**
Syntax:  `chef clean {recipe} [build-configs...]`

The subcommand runs the `bitbake -c cleansstate {recipe}` in each
build-config sub-directories.

This will erase all the files produced during the compilation of the
recipe and the associated sstate-cache files.

### CODE REVIEW

File: chef/chef.py [hunk 1]
```diff
@@ -498,6 +498,50 @@ def build_target(self, build, sdk):
...
+                if not build in BuildConfiguration.ALL:
+                    fatal_error('undefined build:', build)
+
+                if not BuildConfiguration.ALL[build].target():
+                    fatal_error('build has no target:', build)
+
+                filtered_builds.append(BuildConfiguration.ALL[build])
+
+        else:
+            filtered_builds = [ x for x in BuildConfiguration.ALL.values() if x.target() ]
```
Comment (pboettch) [inline chef/chef.py:518]: Should make a function for this loop, it exists several times now.

File: chef/chef.py [hunk 2]
```diff
@@ -498,6 +498,50 @@ def build_target(self, build, sdk):
...
+        try:
+            info('Clean {} for {}'.format(recipe, build.name()))
+
+            directory = build.dir()
+
+            init_script = self.config.layer_dir(ChefCommands.DEFAULT_INIT_BUILD_SCRIPT)
+            if not os.path.isfile(init_script):
+                fatal_error('init-script', init_script, 'not found')
+
+            command_line = 'env bash -c "source {} {} && bitbake -c cleansstate {}"'.format(init_script, directory, recipe)
```
Comment (pboettch) [inline chef/chef.py:534]: same for construction the command-line

**Comment by cpb-** [comment]
Tests for `chef clean` added.
More to come for the repeated constructions.

File: README.md [hunk 3]
```diff
@@ -76,10 +76,17 @@ Each time you do some changes in the menu file, you may need to call:
...
 - `chef update`: if you have modified a commit number or you want to pull the
   latest version of a branch
+
 - `chef generate`: if you have modified a `local.conf` or a `layers`-attribute.
 
 Then `chef build` to restart the compilations.
 
+Another useful sub-command is:
+
+- `chef clean {recipe} [build-configs...]` that will erase all files produced
```
Comment (pboettch) [inline README.md:86]: Est-ce n'est pas `<>` pour les placeholders des command-line-args?

```suggestion
- `chef clean <recipe> [build-configs...]` that will erase all files produced
```

File: test/basic/clean/test [hunk 4]
```diff
@@ -0,0 +1,161 @@
+# `chef clean recipe` fails when `chef init` has not been called.'
+rm -f .chefconfig
```
Comment (pboettch) [inline test/basic/clean/test:2]: Not needed, as test is always run in a clear directory.

File: README.md (hunk 3 above)
Comment (cpb-) [reply README.md:86]: In the current `README.md` file we wrote `chef cook {menu file} [build-configs...]` and `chef init {menu file}`.
Replacing `{ }` by `< >` may be an option, but for another P.R. I think.

File: test/basic/clean/test (hunk 4 above)
Comment (cpb-) [reply test/basic/clean/test:2]: Yes, you're right.

**Pull request merged**